modules/loader/src/main/java/com/opengamma/strata/loader/csv/FullSwapTradeCsvPlugin.java

```diff
@@ -276,19 +276,20 @@ private static SwapLeg parseLeg(
     PayReceive payReceive = LoaderUtils.parsePayReceive(getValue(row, leg, DIRECTION_FIELD));
     PeriodicSchedule accrualSch = parseAccrualSchedule(row, leg);
     PaymentSchedule paymentSch = parsePaymentSchedule(row, leg, accrualSch.getFrequency());
-    NotionalSchedule notionalSch = parseNotionalSchedule(row, leg);
+    Currency currency = Currency.of(getValueWithFallback(row, leg, CURRENCY_FIELD));
 
     Optional<String> knownAmount = findValue(row, leg, KNOWN_AMOUNT_FIELD);
     if (knownAmount.isPresent()) {
       return KnownAmountSwapLeg.builder()
           .payReceive(payReceive)
           .accrualSchedule(accrualSch)
           .paymentSchedule(paymentSch)
-          .currency(notionalSch.getCurrency())
+          .currency(currency)
           .amount(ValueSchedule.of(LoaderUtils.parseDouble(knownAmount.get())))
           .build();
     }
 
+    NotionalSchedule notionalSch = parseNotionalSchedule(row, leg);
```

Comment:
Unless `parseRateCalculationLeg` is reused, this line should probably move into the method

Reply:
Done